Mathlib/LinearAlgebra/ExteriorAlgebra/Basic.lean

```diff
@@ -330,10 +330,115 @@ theorem ιMulti_succ_curryLeft {n : ℕ} (m : M) :
+  refine fun u hu ↦ Submodule.subset_span ?_
+  obtain ⟨f, hfu⟩ := Set.mem_pow.mp hu
+  rw [Set.mem_range, ← hfu]
+  existsi fun i => ιInv (f i).1
+  rw [ιMulti_apply]
+  congr
+  ext i
+  obtain ⟨v, hv⟩ := LinearMap.mem_range.mp (by simp only [SetLike.coe_mem] :
+    (f i).1 ∈ LinearMap.range (ι R (M := M)))
+  rw [← hv, ι_inj]; erw [ι_leftInverse]
+
+/-- Given a linearly ordered family `v` of vectors of `M` and a natural number `n`, produce the
+family of `n`fold exterior products of elements of `v`, seen as members of the exterior algebra.-/
+noncomputable def ιMulti_family (n : ℕ) {I : Type*} [LinearOrder I] (v : I → M) :
+    {s : Finset I // Finset.card s = n} → ExteriorAlgebra R M :=
+  fun ⟨s, hs⟩ ↦ ιMulti R n (fun i => v (Finset.orderIsoOfFin s hs i))
+
+variable {R}
+
 /-- An `ExteriorAlgebra` over a nontrivial ring is nontrivial. -/
 instance [Nontrivial R] : Nontrivial (ExteriorAlgebra R M) :=
   (algebraMap_leftInverse M).injective.nontrivial
 
+/-! Functoriality of the exterior algebra.-/
+
+variable {N : Type u4} {N' : Type u5} [AddCommGroup N] [Module R N] [AddCommGroup N'] [Module R N']
+
+/-- The morphism of exterior algebras induced by a linear map.-/
+def map (f : M →ₗ[R] N) : ExteriorAlgebra R M →ₐ[R] ExteriorAlgebra R N :=
+  CliffordAlgebra.map { f with map_app' := fun _ => rfl }
+
+@[simp]
+theorem map_comp_ι (f : M →ₗ[R] N) : (map f).toLinearMap ∘ₗ ι R = ι R ∘ₗ f :=
+  CliffordAlgebra.map_comp_ι _
+
+@[simp]
+theorem map_apply_ι (f : M →ₗ[R] N) (m : M) : map f (ι R m) = ι R (f m) :=
+  CliffordAlgebra.map_apply_ι _ m
+
+@[simp]
+theorem map_apply_ιMulti {n : ℕ} (f : M →ₗ[R] N) (m : Fin n → M) :
+    map f (ιMulti R n m) = ιMulti R n (f ∘ m) := by
+  rw [ιMulti_apply, ιMulti_apply, map_list_prod]
+  simp only [List.map_ofFn, Function.comp, map_apply_ι]
+
+@[simp]
+theorem map_comp_ιMulti {n : ℕ} (f : M →ₗ[R] N) :
+    (map f).toLinearMap.compAlternatingMap (ιMulti R n (M := M)) =
+    (ιMulti R n (M := N)).compLinearMap f := by
+  ext m
+  exact map_apply_ιMulti _ _
+
+@[simp]
+theorem map_id :
+    map LinearMap.id = AlgHom.id R (ExteriorAlgebra R M) :=
+  CliffordAlgebra.map_id 0
+
+@[simp]
+theorem map_comp_map (f : M →ₗ[R] N) (g : N →ₗ[R] N') :
+    AlgHom.comp (map g) (map f) = map (LinearMap.comp g f) :=
+  CliffordAlgebra.map_comp_map _ _
+
+@[simp]
+theorem ι_range_map_map (f : M →ₗ[R] N) :
+    Submodule.map (AlgHom.toLinearMap (map f)) (LinearMap.range (ι R (M := M))) =
+    Submodule.map (ι R) (LinearMap.range f) :=
+  CliffordAlgebra.ι_range_map_map _
+
+/-- A morphism of modules that is injective and admits a linear section induces an injective
+morphism of exterior algebras.-/
+lemma map_injective {f : M →ₗ[R] N} (hf : ∃ (g : N →ₗ[R] M), g.comp f = LinearMap.id) :
+    Function.Injective (map f) :=
+  let ⟨g, hgf⟩ := hf
+  Function.RightInverse.injective (g := map g) (fun _ ↦ by
+    rw [← AlgHom.comp_apply, map_comp_map, hgf, map_id]; simp only [AlgHom.coe_id, id_eq])
```

Comment:
Will do tomorrow, I really need to go to bed now.

Reply:
I wrote the lemma about left inverses. For the linear maps, I'm keeping the "left inverse" condition as `g.comp f = LinearMap.id` rather than `LeftInverse g f`, because this is the condition that [LinearMap.exists_leftInverse_of_injective](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/Basis/VectorSpace.html#LinearMap.exists_leftInverse_of_injective) gives me in its conclusion, so it makes the lemma easier to use for modules over a field.